src/main/apiClient.ts

```diff
@@ -620,6 +617,48 @@ export default class ApiClient {
     }
   }
 
+  signupWithWebAuthn(params: SignupWithWebAuthnParams, auth?: AuthOptions) {
+    if (navigator.credentials && navigator.credentials.get) {
+      const body = {
+        origin: window.location.origin,
+        clientId: this.config.clientId,
+        friendlyName: params.friendlyName || window.navigator.platform,
+        profile: params.profile,
+        scope: params.scope,
+        redirectUrl: params.redirectUrl
+      }
+
+      const registrationOptionsPromise = this.http.post<RegistrationOptions>('/webauthn/signup-options', { body })
+
+      const credentialsPromise = registrationOptionsPromise.then(response => {
+        const publicKey = encodePublicKeyCredentialCreationOptions(response.options.publicKey)
+
+        return navigator.credentials.create({ publicKey })
+      })
+
+      return Promise.all([registrationOptionsPromise, credentialsPromise])
+        .then(([registrationOptions, credentials]) => {
+          if (!credentials || credentials.type !== publicKeyCredentialType) {
+            throw new Error('Unable to register invalid public key credentials.')
```

Comment:
While it's technically the same, we're going with `return Promise.reject(new Error(...))` in the redirectless PR. This will help to make error handling a little more consistent across the SDK code.

Reply:
```suggestion
            return Promise.reject(new Error('Unable to register public key credentials.'))
```